starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -386,11 +396,14 @@ def from_python(
         calldata: List[int] = []
         for name, cairo_type in type_by_name.items():
             if name not in named_arguments:
-                raise TypeError(f"Input {name} not provided.")
+                raise CairoSerializerException(f"Input {name} not provided.")
 
-            values = self.resolve_type(cairo_type).from_python(
-                cairo_type, name, named_arguments[name]
-            )
+            try:
+                values = self.resolve_type(cairo_type).from_python(
+                    cairo_type, name, named_arguments[name]
+                )
+            except (TypeError, ValueError) as err:
+                raise InvalidDataException(str(err)) from err
```

Comment:
This is still kind of problematic, because some `TypeErrors` would be that way replaced by `InvalidDataException` which extends `ValueError`